tardis/montecarlo/montecarlo.pyx

```diff
@@ -86,61 +87,31 @@ cdef extern from "src/cmontecarlo.h":
 
     void montecarlo_main_loop(storage_model_t * storage, int_type_t virtual_packet_flag, int nthreads, unsigned long seed)
 
-def montecarlo_radial1d(model, runner, int_type_t virtual_packet_flag=0,
-                        int nthreads=4):
-    """
-    Parameters
-    ----------
-    model : `tardis.model_radial_oned.ModelRadial1D`
-        complete model
-    param photon_packets : PacketSource object
-        photon packets
 
-    Returns
-    -------
-    output_nus : `numpy.ndarray`
-    output_energies : `numpy.ndarray`
 
-    TODO
-                    np.ndarray[double, ndim=1] line_list_nu,
-                    np.ndarray[double, ndim=2] tau_lines,
-                    np.ndarray[double, ndim=1] ne,
-                    double packet_energy,
-                    np.ndarray[double, ndim=2] p_transition,
-                    np.ndarray[int_type_t, ndim=1] type_transition,
-                    np.ndarray[int_type_t, ndim=1] target_level_id,
-                    np.ndarray[int_type_t, ndim=1] target_line_id,
-                    np.ndarray[int_type_t, ndim=1] unroll_reference,
-                    np.ndarray[int_type_t, ndim=1] line2level,
-                    int_type_t log_packets,
-                    int_type_t do_scatter
-    """
-    cdef storage_model_t storage
-    cdef np.ndarray[double, ndim=1] packet_nus = model.packet_src.packet_nus
-    storage.packet_nus = <double*> packet_nus.data
-    cdef np.ndarray[double, ndim=1] packet_mus = model.packet_src.packet_mus
-    storage.packet_mus = <double*> packet_mus.data
-    cdef np.ndarray[double, ndim=1] packet_energies = model.packet_src.packet_energies
-    storage.packet_energies = <double*> packet_energies.data
-    storage.no_of_packets = packet_nus.size
+
+cdef initialize_storage_model(model, runner, storage_model_t *storage):
+
+    storage.no_of_packets = model.packet_src.packet_nus.size
+    storage.packet_nus = <double*> PyArray_DATA(model.packet_src.packet_nus)
+    storage.packet_mus = <double*> PyArray_DATA(model.packet_src.packet_mus)
+    storage.packet_energies = <double*> PyArray_DATA(model.packet_src.packet_energies)
+
     # Setup of structure
     structure = model.tardis_config.structure
     storage.no_of_shells = structure.no_of_shells
-    cdef np.ndarray[double, ndim=1] r_inner = structure.r_inner.to('cm').value
-    storage.r_inner = <double*> r_inner.data
-    cdef np.ndarray[double, ndim=1] r_outer = structure.r_outer.to('cm').value
-    storage.r_outer = <double*> r_outer.data
-    cdef np.ndarray[double, ndim=1] v_inner = structure.v_inner.to('cm/s').value
-    storage.v_inner = <double*> v_inner.data
+    storage.r_inner = <double*> PyArray_DATA(structure.r_inner.to('cm').value)
```

Comment:
@orbitfold this is the main thing I'm doing - a much easier way to access the arrays.